### PR TITLE
feat: random normal

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

File: nx/lib/nx/random.ex [hunk 1]
```diff
@@ -413,6 +413,110 @@ defmodule Nx.Random do
...
+      {t, _} when t == :f or t == :bf ->
+        normal_real(key, mean, standard_deviation, opts)
+
+      _ ->
+        raise ArgumentError,
+              "expected float or complex type, got type #{inspect(type)}"
+    end
+  end
+
+  defn normal_real(key, mean, standard_deviation, opts \\ []) do
```
Comment (seanmor5) [inline nx/lib/nx/random.ex:506]: ?

```suggestion
  defnp normal_real(key, mean, standard_deviation, opts \\ []) do
```

**Review by seanmor5 — APPROVED** [review]

File: nx/lib/nx/random.ex (hunk 1 above)
Comment (polvalente) [reply nx/lib/nx/random.ex:506]: Yeap. I had it public for debugging

File: nx/lib/nx/random.ex [hunk 2]
```diff
@@ -413,6 +413,110 @@ defmodule Nx.Random do
...
+
+    * `:type` - a float or complex type for the returned tensor
+
+    * `:shape` - shape of the returned tensor
+
+    * `:names` - the names of the returned tensor
+
+  ## Examples
+
+    iex> key = Nx.Random.key(42)
```
Comment (josevalim) [inline nx/lib/nx/random.ex:429]: Please indent the examples with four spaces for consistency. :)

File: nx/lib/nx/random.ex [hunk 3]
```diff
@@ -413,6 +413,110 @@ defmodule Nx.Random do
...
+        normal_real(key, mean, standard_deviation, opts)
+
+      _ ->
+        raise ArgumentError,
+              "expected float or complex type, got type #{inspect(type)}"
+    end
+  end
+
+  defnp normal_real(key, mean, standard_deviation, opts \\ []) do
+    u = uniform(key, -1, 1, opts)
```
Comment (josevalim) [inline nx/lib/nx/random.ex:507]: Doesn't this mean we include `-1`?

File: nx/lib/nx/random.ex [hunk 4]
```diff
@@ -413,6 +413,110 @@ defmodule Nx.Random do
...
+
+    normal = Nx.sqrt(2) * Nx.erf_inv(u)
+
+    result = standard_deviation * normal + mean
+    Nx.as_type(result, opts[:type])
```
Comment (josevalim) [inline nx/lib/nx/random.ex:512]:
```suggestion
    normal = Nx.sqrt(Nx.type(2, type: opts[:type]) * Nx.erf_inv(u)
    standard_deviation * normal + mean
```

File: nx/lib/nx/random.ex (hunk 3 above)
Comment (polvalente) [reply nx/lib/nx/random.ex:507]: Yes. Although the maximum value is open-ended, so I gotta check if this affects the distribution (I think not, given the 1000 example)

File: nx/lib/nx/random.ex (hunk 2 above)
Comment (josevalim) [reply nx/lib/nx/random.ex:429]: Ah, yes pls.

**Review by josevalim — APPROVED** [review]

**Pull request merged**